lib/chartjs/chart_helpers.rb

```diff
@@ -36,12 +36,16 @@ def chart(klass, data, options)
       script = javascript_tag do
         <<-END.squish.html_safe
         (function() {
+        
           var initChart = function() {
             var data = #{data.to_json};
             var opts = #{options.to_json};
             if (!("animation" in opts)) {
               opts["animation"] = (typeof Modernizr == "undefined") || Modernizr.canvas;
             }
+            
+            canvas && canvas.destroy();
```

Comment:
I think this would work better:

```ruby
window.Chart[#{element_id.to_json}] && window.Chart[#{element_id.to_json}].destroy();
```

Please could you try that?